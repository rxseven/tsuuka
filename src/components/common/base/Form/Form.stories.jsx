import React from 'react';
import { storiesOf } from '@storybook/react';

import Form from './index';

const title = 'Components/Common/Base/Form';

storiesOf(title, module)
  .addParameters({
    info: {
      text: `
  **Usage:** for more information, see [Bootstrap - Forms](https://getbootstrap.com/docs/4.1/components/forms/).
`
    }
  })
  .add('default', () => (
    <Form>
      <input className="form-control" placeholder="Text input" type="text" />
    </Form>
  ));
