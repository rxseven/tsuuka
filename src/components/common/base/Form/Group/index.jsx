import cx from 'classnames';
import PropTypes from 'prop-types';
import exact from 'prop-types-exact';
import React from 'react';

const propTypes = exact({
  children: PropTypes.node.isRequired,
  className: PropTypes.string
});

const defaultProps = {
  className: ''
};

function Group({ children, className }) {
  return <div className={cx('form-group', className)}>{children}</div>;
}

Group.propTypes = propTypes;
Group.defaultProps = defaultProps;

export default Group;
