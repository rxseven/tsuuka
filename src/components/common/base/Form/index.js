import PropTypes from 'prop-types';
import exact from 'prop-types-exact';
import React from 'react';

import Group from './Group';

const propTypes = exact({
  children: PropTypes.node.isRequired,
  className: PropTypes.string
});

const defaultProps = {
  className: ''
};

function Form({ children, className }) {
  return <form className={className}>{children}</form>;
}

Form.propTypes = propTypes;
Form.defaultProps = defaultProps;

Form.Group = Group;

export default Form;
