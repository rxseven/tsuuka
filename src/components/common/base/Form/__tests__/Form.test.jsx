import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import Form from '../index';

describe('<Form />', () => {
  // Arrange
  const props = { children: <div>Form controls</div> };
  const component = <Form {...props} />;

  describe('Unit tests', () => {
    it('should render without crashing', () => {
      // Act
      const wrapper = shallow(component);

      // Assert
      expect(wrapper).toBeDefined();
    });
  });

  describe('Snapshot tests', () => {
    it('should render correctly', () => {
      // Act
      const tree = renderer.create(component).toJSON();

      // Assert
      expect(tree).toMatchSnapshot();
    });
  });
});
