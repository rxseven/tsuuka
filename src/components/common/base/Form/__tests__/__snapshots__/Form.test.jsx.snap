// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> Snapshot tests should render correctly 1`] = `
<form
  className=""
>
  <div>
    Form controls
  </div>
</form>
`;
